compiler/parser/chapel.lex

```diff
@@ -409,6 +409,17 @@ static int processToken(yyscan_t scanner, int t) {
       captureString.push_back(' ');
     }
 
+    // this is here to remove extra space added between
+    // the end of a recognized keyword from the list below
+    // and a closing paren or dot.
+    // e.g print maxIndex(A: [?D]) instead of maxIndex(A: [?D] )
+    if (t == TRP || t == TDOT) {
+      auto last = captureString.back();
+      if (last == ' ' && captureString.length() > 1) {
```

Comment:
Recommend

```
if (captureString.length() > 1 &&
    captureString.back() == ' ') {
```

since it will be a memory error to get captureString.back() if the size is 0 (so should check the size first).